dev-db/etcd-wrapped/files/etcd-wrapped.ebuild

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2016 CoreOS, Inc.. All rights reserved.
+# Distributed under the terms of the GNU General Public License v2
+# $Header:$
+#
+
+EAPI=6
+
+inherit systemd
+
+DESCRIPTION="etcd Wrapper"
+HOMEPAGE="https://github.com/coreos/etcd"
+KEYWORDS="amd64 arm64"
+
+LICENSE="Apache-2.0"
+IUSE="etcdctl"
+
+VERSION=${PV}
+VERSION=${VERSION/_alpha/-alpha.}
+VERSION=${VERSION/_beta/-beta.}
+
+case ${VERSION} in
+0*)
+	SLOT="0"
+	;;
+2*)
+	SLOT="2"
+	;;
+3*)
+	SLOT="3"
+	;;
+*)
+	die "Unknown etcd version ${VERSION}"
+	;;
+esac
+
+
+SLOT_MANGLED=${SLOT/0/}
+IMG_FILENAME="etcd-v${VERSION}-linux-${ARCH}.aci"
+PUBKEY_FILENAME="etcd-v${VERSION}_pubkey.gpg"
+IMG_PREFIX="coreos.com/etcd"
+IMG_PUBKEY="${DISTDIR}/${PUBKEY_FILENAME}"
+RKT="rkt"
+VERSION_DIR=/usr/lib/coreos/versions/
+IMG_VERSION_FILE="${VERSION_DIR}/etcd${SLOT_MANGLED}"
+IMG_VERSION_DIR="${VERSION_DIR}/etcd${SLOT_MANGLED}.d"
+ENV_PREFIX="ETCD"
+
+
+SRC_URI="\
+	https://github.com/coreos/etcd/releases/download/v${VERSION}/${IMG_FILENAME} \
+	https://github.com/coreos/etcd/releases/download/v${VERSION}/${IMG_FILENAME}.asc \
+	https://coreos.com/dist/pubkeys/app-signing-pubkey.gpg -> ${PUBKEY_FILENAME} \
+"
+
+DEPEND=""
+RDEPEND="\
+	app-emulation/rkt[rkt_stage1_fly]
+	!dev-db/etcd
+"
+
+S=${WORKDIR}
+
+
+
+pkg_preinst() {
+#	${RKT} trust \
+#		--prefix="${IMG_PREFIX}" \
+#		--skip-fingerprint-review \
+#		${IMG_PUBKEY} \
+#		|| die "rkt trust failed"
+#
+#	${RKT} fetch \
+#		"${DISTDIR}/${IMG_FILENAME}" \
+#		> fetch.log || die "rkt fetch failed"
+#
+#	echo ${ENV_PREFIX}_IMG_FACTORY_HASH=$(tail -n1 fetch.log) >> "${D}/${IMG_VERSION_FILE}"
+	echo ${ENV_PREFIX}_IMG="/usr/lib/coreos/versions/etcd${SLOT_MANGLED}.d/${IMG_FILENAME}" >> "${D}/${IMG_VERSION_FILE}"
```

Comment:
etcd${SLOT}.d would only ever contain a single aci at a time. Why bother with the extra indirection when it could just be etcd${SLOT}.aci in the first place?